### ISSUE TITLE
Expanding array

### ISSUE THREAD

**Issue by yosiat** [tjmehta/keypather#25]
Hi,

`keypath.expand` behavior is pretty weird with expanding an array.

The output  of this:
```
keypath.expand(keypath.flatten({
   notes: ["hello", "this"]
}));
```

is `{ "0": "hello", "1": "this" }`

**Comment by tjmehta** [comment]
Interesting is it also missing the “notes” key altogether?

**Comment by yosiat** [comment]
@tjmehta added tests for this in - https://github.com/tjmehta/keypather/pull/26
and it's not missing the `notes`, it is returning
`{ "notes":  { "0": "hello", "1": "this" } }`

**Comment by tjmehta** [comment]
published keypather@v3.0.0 lmk if you still have problems
